src/util/contextOf.ts

```diff
@@ -0,0 +1,10 @@
+import { Context, Path } from "../types";
+
+type ReturnType = {
+  (param:Context): Context;
+  (param:Path): Path;
+  (param: Path | Context) : Path | Context
+}
```

Comment:
Can you explain this? I don't quite understand.

Reply:
An explicit return type is added here to allow the compiler to infer the type of results in the calling functions. If we don't do it, the return type would be `Path | Context`.  
An example from `importHtml.ts` - 
[importCursor](https://github.com/cybersemics/em/blob/adab4290f2a0c59d6fc53f52a3dd8040b3f21dba/src/util/importHTML.js#L98) can be `Path|Context`. But [here](https://github.com/cybersemics/em/blob/adab4290f2a0c59d6fc53f52a3dd8040b3f21dba/src/util/importHTML.js#L182), we're assuming its type to be `Path`, which results in a compile-time error.